monkey/infection_monkey/payload/payload.py

```diff
@@ -0,0 +1,21 @@
+from abc import abstractmethod
+
+import infection_monkey.payload
+from infection_monkey.utils.plugins import Plugin
+
+
+class Payload(Plugin):
+    @staticmethod
+    def base_package_file():
+        return infection_monkey.payload.__file__
+
+    @staticmethod
+    def base_package_name():
+        return infection_monkey.payload.__package__
+
+    @abstractmethod
+    def run_payload(self):
+        raise NotImplementedError()
+
+    def should_run(self):
```

Comment:
Why is this here? We'll remove it later, because it's not the responsibility of plugin to decide whether it should run? Add a todo then? Or return True to adhere to the Plugin interface?

Reply:
The point of `should_run` is to see if we have enabled the payload in the configuration. For this case I think either we raise `Not ImplementedError` or just leave it as it is.

Reply:
@ilija-lazoroski @VakarisZ @shreyamalviya It's the master's job to decide whether or not the payload should run. Neither the puppet nor the payload should be making any such decision. The puppet does what its told, full stop.